bentoml/_internal/server/grpc/interceptors/access_log.py

```diff
@@ -0,0 +1,43 @@
+import logging
+from timeit import default_timer
+
+from opentelemetry import trace
+
+from bentoml._internal.configuration.containers import BentoMLContainer
+from bentoml._internal.server.grpc.interceptors import AsyncServerInterceptor
+
+
+class AccessLogInterceptor(AsyncServerInterceptor):
+    def __init__(self) -> None:
+        self.logger = logging.getLogger("bentoml.access")
+
+    async def intercept(
+        self,
+        method,
+        request,
+        context,
+        method_name: str,
+    ) -> None:
+        tracer_provider = BentoMLContainer.tracer_provider.get()
+        tracer = tracer_provider.get_tracer("opentelemetry.instrumentation.grpc")
+        span = tracer.start_span("grpc")
+
+        start = default_timer()
+        with trace.use_span(span, end_on_exit=True):
+            response = await method(request, context)
+        latency = max(default_timer() - start, 0)
+
+        request_info: str = f"api_name={request.api_name},type={str(response.contents.WhichOneof('kind'))[:-6]},size={request.contents.ByteSize()}"
```

Comment:
We shouldn't getting kind like this, since kind can be None.
In that case we should raise an error via context.